docs/guide/load-when-visible.md

```diff
@@ -0,0 +1,349 @@
+    <div>Loading...</div>
+  </svelte:fragment>
+
+  {#each permissions as permission}
+    <!-- ... -->
+  {/each}
+</WhenVisible>
+```
+
+== Svelte 5
+
+```svelte
+<script>
+  import { WhenVisible } from '@inertiajs/svelte'
+
+  let { permissions } = $props()
+</script>
+
+<WhenVisible data="permissions">
+  {#snippet fallback()}
+    <div>Loading...</div>
+  {/snippet}
+
+  {#each permissions as permission}
+    <!-- ... -->
+  {/each}
+</WhenVisible>
+```
+
+:::
+
+If you'd like to load multiple props when an element becomes visible, you can provide an array to the `data` prop.
+
+:::tabs key:frameworks
+== Vue
+
+```vue
+<script setup>
+import { WhenVisible } from '@inertiajs/vue3'
+</script>
+
+<template>
+  <WhenVisible :data="['teams', 'users']">
+    <template #fallback>
+      <div>Loading...</div>
+    </template>
+    <!-- Props are now loaded -->
+  </WhenVisible>
+</template>
+```
+
+== React
+
+```jsx
+import { WhenVisible } from '@inertiajs/react'
+
+export default () => (
+  <WhenVisible data={['teams', 'users']} fallback={<div>Loading...</div>}>
+    <ChildComponent />
+  </WhenVisible>
+)
+```
+
+== Svelte 4
+
+```svelte
+<script>
+  import { WhenVisible } from '@inertiajs/svelte'
+
+  export let teams
+  export let users
+</script>
+
+<WhenVisible data={['teams', 'users']}>
+  <svelte:fragment slot="fallback">
+    <div>Loading...</div>
+  </svelte:fragment>
+
+  <!-- Props are now loaded -->
+</WhenVisible>
+```
+
+== Svelte 5
+
+```svelte
+<script>
+  import { WhenVisible } from '@inertiajs/svelte'
+
+  let { teams, users } = $props()
+</script>
+
+<WhenVisible data={['teams', 'users']}>
+  {#snippet fallback()}
+    <div>Loading...</div>
+  {/snippet}
+  
+  <!-- Props are now loaded -->
+</WhenVisible>
+```
+:::
+
+## Loading before visible
+
+If you'd like to start loading data before the element is visible, you can provide a value to the `buffer` prop. The buffer value is a number that represents the number of pixels before the element is visible.
+
+:::tabs key:frameworks
+== Vue
+
+```vue
+<script setup>
+import { WhenVisible } from '@inertiajs/vue3'
+</script>
+
+<template>
+  <WhenVisible data="permissions" :buffer="500">
+    <template #fallback>
+      <div>Loading...</div>
+    </template>
+
+    <div v-for="permission in permissions">
+      <!-- ... -->
+    </div>
+  </WhenVisible>
+</template>
+```
+
+== React
+
+```jsx
+import { WhenVisible } from '@inertiajs/react'
+
+export default () => (
+  <WhenVisible data="permissions" buffer={500} fallback={<div>Loading...</div>}>
+    <PermissionsChildComponent />
+  </WhenVisible>
+)
+```
+
+== Svelte 4
+
+```svelte
+<script>
+  import { WhenVisible } from '@inertiajs/svelte'
+
+  export let permissions
+</script>
+
+<WhenVisible data="permissions" buffer={500}>
+  <svelte:fragment slot="fallback">
+    <div>Loading...</div>
+  </svelte:fragment>
+  
+  {#each permissions as permission}
+    <!-- ... -->
+  {/each}
+</WhenVisible>
+
```

Comment:
the end of the svelte code block for svelte 4 is missing. Therefore the text explaining the 'as' appears only when selecting svelte 5.
![image](https://github.com/user-attachments/assets/82627e84-8e50-4b9b-a366-a646780a2390)
![image](https://github.com/user-attachments/assets/6a16b726-c2ca-4775-a65c-beef096e6c50)